File: THCS/2021_2022/DaNang_2122/Bai1/SNT.cpp
#include <iostream>
#include <cmath>
using namespace std;

// Hàm kiểm tra một số có phải số nguyên tố hay không
bool isPrime(int n) {
    if (n <= 1) return false;
    if (n == 2 || n == 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }
    return true;
}

int main() {
    int a;
    cout << "Nhập số nguyên dương a: ";
    cin >> a;

    int b = 1;
    while (!isPrime(a + b)) {
        b++;
    }

    cout << "Số nguyên dương b nhỏ nhất là: " << b << endl;
    return 0;
}

File: THCS/2021_2022/DaNang_2122/Bai1/SNT.py
import math

# Hàm kiểm tra một số có phải số nguyên tố hay không
def is_prime(n):
    if n <= 1:
        return False
    if n == 2 or n == 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    for i in range(5, int(math.sqrt(n)) + 1, 6):
        if n % i == 0 or n % (i + 2) == 0:
            return False
    return True

# Nhập số nguyên dương a từ bàn phím
a = int(input("Nhập số nguyên dương a: "))

b = 1
while not is_prime(a + b):
    b += 1

# In ra số nguyên dương b nhỏ nhất
print(f"Số nguyên dương b nhỏ nhất là: {b}")
